### PR TITLE
Allow typenum to be built without the standard library.

### PR DESCRIPTION
**Pull request by gereeter**
The build.rs stuff is hacky, but should work.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -34,6 +34,11 @@ assert_eq!(<X as Integer>::to_i32(), 7);
 
 Documented in each module is the full list of type operators implemented.
 */
+#![cfg_attr(feature="no_std", feature(core, no_std))]
+#![cfg_attr(feature="no_std", no_std)]
+#[cfg(feature="no_std")]
+extern crate core as std;
```
Comment (llogiq) [inline src/lib.rs]: Why not use `core::cmp::Ordering` directly? It should be available even with `std`.

Comment (gereeter) [inline src/lib.rs]: Using `core` at all is impossible on stable. Also, there are many other uses of `std` besides just `Ordering`.

Comment (gereeter) [inline src/lib.rs]: This statement makes the whole crate, not just this module, see `std` to mean `core`.

Comment (llogiq) [inline src/lib.rs]: I see. Clever trick, that. :smile:

**Comment by paholg** [comment]
I like this.

It would be nice for core to be stabilized so we could do away with `extern crate core as std`, but not much to be done about it now.

I assume it's okay for the build script to use std?

Also, to stick with semver, this should bump us up to 1.1.0?

**Comment by llogiq** [comment]
Yeah, I just tested it and found no ill effects. I think it's safe to merge; we can do the version bump (and docs!) apart from the PR.

**Pull request merged**